src/coreclr/jit/lowerxarch.cpp

```diff
@@ -850,372 +850,402 @@ void Lowering::LowerCast(GenTree* tree)
+                        if (srcType == TYP_FLOAT)
+                        {
+                            maxFloatSimdVal->f32[0] = 2147483648.0f;
+                            convertIntrinsic        = NI_X86Base_ConvertToInt32WithTruncation;
+                        }
+                        else
+                        {
+                            maxFloatSimdVal->f64[0] = 2147483648.0;
+                            convertIntrinsic        = NI_X86Base_ConvertToInt32WithTruncation;
+                        }
+                        break;
+                    }
+                    case TYP_UINT:
+                    {
+                        dstBaseType      = CORINFO_TYPE_UINT;
+                        maxIntegralValue = comp->gtNewIconNode(static_cast<ssize_t>(UINT32_MAX));
+                        if (srcType == TYP_FLOAT)
+                        {
+                            maxFloatSimdVal->f32[0] = 4294967296.0f;
+                            convertIntrinsic        = TargetArchitecture::Is64Bit
+                                                          ? NI_X86Base_X64_ConvertToInt64WithTruncation
+                                                          : NI_X86Base_ConvertToVector128Int32WithTruncation;
+                        }
+                        else
+                        {
+                            maxFloatSimdVal->f64[0] = 4294967296.0;
+                            convertIntrinsic        = TargetArchitecture::Is64Bit
+                                                          ? NI_X86Base_X64_ConvertToInt64WithTruncation
+                                                          : NI_X86Base_ConvertToVector128Int32WithTruncation;
+                        }
+                        break;
+                    }
+                    case TYP_LONG:
+                    {
+                        dstBaseType      = CORINFO_TYPE_LONG;
+                        maxIntegralValue = comp->gtNewLconNode(INT64_MAX);
+                        if (srcType == TYP_FLOAT)
+                        {
+                            maxFloatSimdVal->f32[0] = 9223372036854775808.0f;
+                            convertIntrinsic        = NI_X86Base_X64_ConvertToInt64WithTruncation;
+                        }
+                        else
+                        {
+                            maxFloatSimdVal->f64[0] = 9223372036854775808.0;
+                            convertIntrinsic        = NI_X86Base_X64_ConvertToInt64WithTruncation;
+                        }
+                        break;
+                    }
+                    case TYP_ULONG:
+                    {
+                        dstBaseType      = CORINFO_TYPE_ULONG;
+                        maxIntegralValue = comp->gtNewLconNode(static_cast<int64_t>(UINT64_MAX));
+                        if (srcType == TYP_FLOAT)
+                        {
+                            maxFloatSimdVal->f32[0] = 18446744073709551616.0f;
```

Comment:
Nit: A comment explicitly stating what these constants are supposed to be (rounded value of `UINT64_MAX`? `UINT64_MAX + 1`?) Would be good so anyone looking at this knows what to do when changing it and can recognize any errors